src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -47,59 +75,152 @@ private CoseHeaderMap(bool isReadOnly)
+        /// <exception cref="InvalidOperationException">Decodification failed, see the inner exception for the exact exception that was thrown.
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public string GetValueAsString(CoseHeaderLabel label) => _headerParameters[label].GetValueAsString();
 
+        /// <summary>
+        /// Gets the CBOR encoded value associated with the specified label, as a byte string.
+        /// </summary>
+        /// <param name="label">The label of the value to get.</param>
+        /// <returns>The decoded value associated with the specified label, as a byte string.</returns>
+        /// <exception cref="InvalidOperationException">Decodification failed, see the inner exception for the exact exception that was thrown.
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public byte[] GetValueAsBytes(CoseHeaderLabel label) => _headerParameters[label].GetValueAsBytes();
 
+        /// <summary>
+        /// Gets the CBOR encoded value associated with the specified label, as a byte string.
+        /// </summary>
+        /// <param name="label">The label of the value to get.</param>
+        /// <param name="destination">The buffer in which to write the decoded value.</param>
+        /// <returns>The number of bytes written to <paramref name="destination" />.</returns>
+        /// <exception cref="ArgumentException"><paramref name="destination"/> is too small to hold the value.</exception>
+        /// <exception cref="InvalidOperationException">Decodification failed, see the inner exception for the exact exception that was thrown.
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public int GetValueAsBytes(CoseHeaderLabel label, Span<byte> destination) => _headerParameters[label].GetValueAsBytes(destination);
 
+        /// <summary>
+        /// Adds the specified key and value to the dictionary.
+        /// </summary>
+        /// <param name="key">The key of the element to add.</param>
+        /// <param name="value">The value of the element to add.</param>
+        /// <exception cref="InvalidOperationException">The <see cref="CoseHeaderMap"/> is read-only.</exception>
+        /// <exception cref="ArgumentException">The property is set and the encoded bytes in the specified <see cref="CoseHeaderValue"/> contain trailing data or more than one CBOR value.</exception>
         public void Add(CoseHeaderLabel key, CoseHeaderValue value)
         {
             ValidateIsReadOnly();
             ValidateInsertion(key, value);
             _headerParameters.Add(key, value);
         }
 
+        /// <summary>
+        /// Adds the specified value to the <see cref="ICollection{T}"/> with the specified key.
```

Comment:
Is `<see cref="ICollection{T}" />` right here? Should it be `<see cref="CoseHeaderMap"/>` like the other `Add` members?

Reply:
I wrote ICollection<T> here because that's what Dictionary<Tk, Tv> does for that method, but Dictionary does an explicit interface implementation and we don't. 
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.system-collections-generic-icollection-system-collections-generic-keyvaluepair-tkey-tvalue---add?view=net-7.0
I guess it should be as you suggest.

Reply:
Dictionary's docs follow an older style where types were linked a lot.  The new style is to not link to types, so 

```suggestion
        /// Adds the specified value to the header map with the specified key.
```